src/data/messages.js

```diff
@@ -0,0 +1,45 @@
+import { defineMessages } from '@edx/frontend-platform/i18n';
+
+const messages = defineMessages({
+  all_threads: {
+    id: 'discussions.navigation-bar.filter.all-threads',
+    defaultMessage: 'All threads',
+    description: 'Option in dropdown to filter to all threads',
+  },
+  all_topics: {
+    id: 'discussions.navigation-bar.filter.all-topics',
+    defaultMessage: 'Topics',
+    description: 'Option in dropdown to view all topics',
+  },
+  my_threads: {
+    id: 'discussions.navigation-bar.filter.my-threads',
+    defaultMessage: 'My threads',
+    description: 'Option in dropdown to filter to all a user\'s threads',
+  },
+  search_results: {
+    id: 'discussions.navigation-bar.filter.search-results',
+    defaultMessage: '{resultCount} results',
+  },
+  sorted_by: {
+    id: 'discussions.navigation-bar.sort.message',
+    defaultMessage: 'Sorted by {sortBy}',
+  },
+  sort_by_course_structure: {
+    id: 'discussions.navigation-bar.sort.course-structure',
+    defaultMessage: 'Course Structure',
+  },
+  sort_by_last_activity: {
+    id: 'discussions.navigation-bar.sort.last-activity',
+    defaultMessage: 'Recent activity',
+  },
+  sort_by_comment_count: {
+    id: 'discussions.navigation-bar.sort.comment-count',
+    defaultMessage: 'Most activity',
+  },
+  sort_by_vote_count: {
+    id: 'discussions.navigation-bar.sort.vote-count',
+    defaultMessage: 'Most votes',
+  },
+});
+
+export default messages;
```

Comment:
Why is this in the data folder? Move it to the navigation bar component.

Reply:
Aside from `all_topics`, `all_threads` and `my_threads`, none of these are used by the navigation bar component, and in some cases, these are used in multiple places elsewhere, which is why I moved it into `data`. I'll see if I can more accurately rename the `id` for some of these.

Reply:
If strings are used twice, define them twice. If they are used in two different contexts they need to have two different IDs.